src/components/CommentField.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useRef, useState, FormEvent, useEffect } from 'react';
+import { EnrichedActivity } from 'getstream';
+
+import { Avatar } from './Avatar';
+import { Button } from './Button';
+import { Textarea, TextareaProps } from './Textarea';
+import { inputValueFromEvent } from '../utils';
+import { useTranslationContext, FeedManager } from '../Context';
+import { DefaultAT, DefaultUT } from '../Context/StreamApp';
+
+export type CommentFieldProps<UT extends DefaultUT = DefaultUT, AT extends DefaultAT = DefaultAT> = {
+  activity: EnrichedActivity<UT, AT>;
+  onAddReaction: FeedManager<UT, AT>['onAddReaction'];
+  image?: string;
+  kind?: string;
+  onSuccess?: () => void;
+  placeholder?: string;
+  trigger?: TextareaProps['trigger'];
+};
+
+export const CommentField = <UT extends DefaultUT = DefaultUT, AT extends DefaultAT = DefaultAT>({
+  activity,
+  onAddReaction,
+  onSuccess,
+  image,
+  placeholder,
+  trigger,
+}: CommentFieldProps<UT, AT>) => {
+  const { t } = useTranslationContext();
+  const textareaReference = useRef<HTMLTextAreaElement>();
+  const [text, setText] = useState<string>();
+
+  const handleFormSubmit = async (event: FormEvent<HTMLFormElement> | KeyboardEvent) => {
+    event.preventDefault();
+
+    if (!text) return;
+
+    try {
+      await onAddReaction('comment', activity, {
+        text,
+      });
+    } catch (error) {
+      console.error(error);
+    }
+
+    setText('');
+    onSuccess?.();
+  };
+
+  const handleFormSubmitKey = (event: KeyboardEvent) => {
+    const { current: textarea } = textareaReference;
+    if (event.key === 'Enter' && textarea?.nextSibling === null) {
+      handleFormSubmit(event);
+    }
+  };
+
+  useEffect(() => {
+    if (!textareaReference.current) return;
+    textareaReference.current.addEventListener('keydown', handleFormSubmitKey);
+
+    return () => textareaReference.current?.removeEventListener('keydown', handleFormSubmitKey);
+  }, []);
+
+  return (
+    <form onSubmit={handleFormSubmit} className="raf-comment-field">
+      <Avatar image={image} circle size={39} />
+      <div className="raf-comment-field__group">
+        <Textarea
+          rows={1}
+          value={text}
+          placeholder={placeholder ?? t('Start Typing...')}
+          onChange={(event) => setText((pv) => inputValueFromEvent<HTMLTextAreaElement>(event) ?? pv)}
```

Comment:
probably a weird edge case that was here but this way it's forcing a state update even if inputValueFromEvent return null or undef

Reply:
I've [tested it before implementing](https://codesandbox.io/s/morning-sea-iv13f) and I'm positive that if you return previous value (as unchanged) it does not trigger  update.

Reply:
interesting, good example 👍